File: Management/Models/CanvasModels/Assignments/CanvasAssignmentOverride.cs
namespace CanvasModel.Assignments;

public record CanvasAssignmentOverride
(

  [property: JsonPropertyName("id")]
  ulong Id,

  [property: JsonPropertyName("assignment_id")]
  ulong AssignmentId,

  [property: JsonPropertyName("course_section_ids")]
  ulong CourseSectionId,

  [property: JsonPropertyName("title")]
  string Title,

  [property: JsonPropertyName("student_ids")]
  IEnumerable<ulong>? StudentIds = null,

  [property: JsonPropertyName("group_id")]
  ulong? GroupId = null,

  [property: JsonPropertyName("due_at")]
  DateTime? DueAt = null,

  [property: JsonPropertyName("all_day")]
  bool? AllDay = null,

  [property: JsonPropertyName("all_day_date")]
  DateTime? AllDayDate = null,

  [property: JsonPropertyName("unlock_at")]
  DateTime? UnlockAt = null,

  [property: JsonPropertyName("lock_at")]
  DateTime? LockAt = null
);

File: nextjs-pages/src/models/canvas/assignments/canvasAssignmentOverride.ts
export interface CanvasAssignmentOverride {
  id: number;
  assignment_id: number;
  course_section_id: number;
  title: string;
  student_ids?: number[];
  group_id?: number;
  due_at?: string; // ISO 8601 date string
  all_day?: boolean;
  all_day_date?: string; // ISO 8601 date string
  unlock_at?: string; // ISO 8601 date string
  lock_at?: string; // ISO 8601 date string
}
